src/App.js

```diff
@@ -28,14 +42,16 @@ const App = () => {
       <div className="App">
         <Switch>
           <Route exact path="/">
-            <List />
+            <Home setToken={setToken} />
+          </Route>
+          <Route exact path="/list">
+            {token ? <List token={token} /> : <Redirect to="/" />}
           </Route>
-
           <Route exact path="/add-item">
             <AddItem />
           </Route>
         </Switch>
-        <NavBar />
+        {token ? <NavBar /> : null}
```

Comment:
Another way to write this in a more concise way:
`{token && <NavBar />}`

Reply:
@Sky shared a great explainer on the [logical AND operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_AND).

Reply:
Thank you for that reminder, Yenly. I just implemented the suggestion. Thank you for including the MDN link that Sky shared, too!